plugins/imfile/imfile.c

```diff
@@ -2589,76 +2686,108 @@ fen_printevent(int event, char *pname)
 	if (event & MOUNTEDOVER) {
 		DBGPRINTF(" MOUNTEDOVER");
 	}
-	DBGPRINTF("\n");
 }
 
-#if false
-/* function not used yet, will be needed for wildcards later */
 static rsRetVal
-fen_monitorDirectory(lstn_t *pLstn)
+fen_removeFile(lstn_t *pLstn)
 {
-	struct file_obj *fobjp = &pLstn->pfinf->fobj;
-	int port = pLstn->pfinf->port;
-	struct stat statFile;
+	uchar statefile[MAXFNAME];
+	uchar toDel[MAXFNAME];
+	int bDoRMState;
+	uchar *statefn;
+	int dirIdx;
 	DEFiRet;
 
-	DBGPRINTF("imfile: fen_monitorDirectory: %s\n", fobjp->fo_name);
+	/* Get correct dirindex from basedir! */
+	dirIdx = dirsFindDir(pLstn->pszDirName);
+	if (dirIdx == -1) {
+		/* Add error processing as required, file may have been deleted/moved. */
+		errmsg.LogError(1, RS_RET_SYS_ERR, "fen_removeFile: Failed to remove file, "
+			"unknown directory index for dir '%s'\n", pLstn->pszDirName);
+		ABORT_FINALIZE(RS_RET_SYS_ERR);
+	}
 
-	/* Get Dir Stats */
-	if (stat(fobjp->fo_name, &statFile) == 0 && S_ISDIR(statFile.st_mode)) {
-		DBGPRINTF("imfile: fen_monitorDirectory: %s is a valid directory, associate port\n"
-			, fobjp->fo_name, errno);
+	DBGPRINTF("fen_removeFile removing listener for '%s', dirIdx %d\n",
+		pLstn->pszFileName, dirIdx);
+	if(	pLstn->bRMStateOnDel) {
+		statefn = getStateFileName(pLstn, statefile, sizeof(statefile), NULL);
+		/* Get full path and file name */
+		getFullStateFileName(statefn, toDel, sizeof(toDel));
+		bDoRMState = 1;
 	} else {
-		DBGPRINTF("imfile: fen_monitorDirectory: Failed to stat directory: %s - errno %d\n"
-			, fobjp->fo_name, errno);
-		ABORT_FINALIZE(RS_RET_FILE_NO_STAT);
+		bDoRMState = 0;
 	}
 
-	/* Register file event */
-	fobjp->fo_atime = statFile.st_atim;
-	fobjp->fo_mtime = statFile.st_mtim;
-	fobjp->fo_ctime = statFile.st_ctim;
-	if (port_associate(port, PORT_SOURCE_FILE, (uintptr_t)fobjp,
-				pLstn->pfinf->events, (void *)pLstn) == -1) {
-		/* Add error processing as required, file may have been deleted/moved. */
-		errmsg.LogError(1, RS_RET_SYS_ERR, "imfile: fen_monitorDirectory: Failed to register directory: "
```

Comment:
This is called incorrectly. I do not know what the ```1``` as first parameter means. Errno should go here, with no explicit reference to errno later on.

Reply:
The errmsg.LogError was copied from do_inotify.

Reply:
so we have two places to fix? ;-)

Reply:
Fixed it in fen code. 
Other places are in inotify code. 
Should be done in another commit.